### DISCUSSION TITLE
Can new_window run a command other than the current shell when using ssh for cwd?

### DISCUSSION THREAD

**Discussion by groves** [kovidgoyal/kitty#5087] [Q&A] [answered]
I've got [this kitten](https://github.com/groves/catherd/blob/main/edit.py#L25) that pops an overlay window with a file finder in the current directory. It works well on my local machine.

When using the ssh kitten, that `cwd_from` argument gets it to ssh into the current host before opening the window, which is super cool. It takes a little bit before my shell prompt is ready through ssh though. If I send the command into the new window from kitty immediately using paste_text, it pastes before the shell is ready and doesn't execute. I've added [this timer hack](https://github.com/groves/catherd/blob/main/edit.py#L72) to give the shell time to get its act together. 

That's obviously unsatisfying as it's fragile time-based code. Is there a way I could run a command other than the shell when kitty sshes in to do my fuzzy finding? Or is there a way to wait till the moment the shell is ready to send my text?

**Comment by kovidgoyal** [comment, marked as answer]
Dont use cwd_from. Instead detect when its an SSH window using
ssh_kitten_cmdline. Then modify the cmdline using set_cwd_in_cmdline()
and add whatever command you want executed to the cmdline and launch
that.

**Reply by groves** [reply]
[Worked like a charm!](https://github.com/groves/catherd/blob/3543c57a5e2caf0822f418ba67abe2718772653e/edit.py#L26) Thanks!
